File: solutions/objective-c/bob/1/Bob.m
#import "Bob.h"

@implementation Bob

-(NSString *)hey:(NSString *)statement {
  if ([statement isEqualToString:@""]) {
    return @"Fine, be that way.";
  } else if ([statement hasSuffix:@"?"]) {
    return @"Sure.";
  } else if ([statement isEqualToString:[statement uppercaseString]]) {
    return @"Woah, chill out!";
  } else {
    return @"Whatever.";
  }
}

@end

File: solutions/swift/bob/1/Sources/Bob/Bob.swift
import Foundation

class Bob {
    class func hey(_ remark: String) -> String {
        let message = remark.trimmingCharacters(in: .whitespaces)
        let noWords = message.uppercased() == message.lowercased()
        
        if message.count == 0 {
            return "Fine. Be that way!"
        }
        if !noWords && message.uppercased() == message {
            return "Whoa, chill out!"
        }
        if message.hasSuffix("?") {
            return "Sure."
        }
        return "Whatever."
    }
}
